File: Python/706A-berutaxi.py
vasiliy1, vasiliy2 = map(int, input().split())
n = int(input())

times = list()
for i in range(n):
    taxi_x, taxi_y, taxi_v = map(int, input().split())
    point1 = (vasiliy1, vasiliy2)
    point2 = (taxi_x, taxi_y)
    d = ((point2[0]-point1[0])**2 + (point2[1] - point1[1])**2) ** 0.5
    times.append(d/taxi_v)

print("{:.20f}".format(min(times)))

File: C++/A/706A-berutaxi.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    long double vasiliy1, vasiliy2, n, taxi_x, taxi_y, taxi_v;
    cin >> vasiliy1 >> vasiliy2;
    cin >> n;
    priority_queue<long double, vector<long double>, greater<long double>> times;
    while (n--)
    {
        cin >> taxi_x >> taxi_y >> taxi_v;
        long double d = hypot(taxi_x - vasiliy1, taxi_y - vasiliy2);
        times.push(d / taxi_v);
    }
    cout << fixed << setprecision(20) << times.top();
}